### PR TITLE
fix issue 20901: etaCollapse context bound type

### PR DESCRIPTION
**Pull request by bishabosha**
fixes https://github.com/scala/scala3/issues/20901

### CODE REVIEW

**Review by odersky — APPROVED** [review]
LGTM

File: tests/pos/i20901/Foo.tastycheck
Comment (hamzaremmal) [inline tests/pos/i20901/Foo.tastycheck:1]: It looks like this test is breaking the `Scala 2 Library TASTy` test in the latest nightly: https://github.com/scala/scala3/actions/runs/9771174027/job/26973528590

Comment (bishabosha) [reply tests/pos/i20901/Foo.tastycheck:1]: The failure is in the TASTy output, but the difference doesn't seem related to this PR - tree 68 changed - the inferred result type
```diff
-68:         SHAREDtype 44
+68:         TYPEREF 17 [Nothing]
+70:           TERMREFpkg 2 [scala]
```

Comment (bishabosha) [reply tests/pos/i20901/Foo.tastycheck:1]: I guess we could possibly update the test to remove inference, or just stop the tasty checking

Comment (bishabosha) [reply tests/pos/i20901/Foo.tastycheck:1]: or test with decompilation rather than printing tasty?

Comment (bishabosha) [reply tests/pos/i20901/Foo.tastycheck:1]: see https://github.com/scala/scala3/pull/21001

**Pull request merged**